File: src/main/java/com/rss/batch/RssSubscriptionWriter.java
package com.rss.batch;

import com.rss.clients.MessagingClient;
import com.rss.model.RssUpdate;
import org.springframework.batch.item.ItemWriter;

import java.util.List;

public class RssSubscriptionWriter implements ItemWriter<List<RssUpdate>> {

    private MessagingClient messagingClient;

    public RssSubscriptionWriter(MessagingClient messagingClient) {
        this.messagingClient = messagingClient;
    }
    @Override
    public void write(List<? extends List<RssUpdate>> list) throws Exception {
        // for now just log em
        for (List<RssUpdate> updateList : list) {
            for (RssUpdate update : updateList) {
                messagingClient.sendRssUpdate(update);
            }
        }
        System.out.println(list);
    }
}

File: src/main/java/com/rss/batch/RssSubscriptionWriter.kt
package com.rss.batch

import com.rss.clients.MessagingClient
import com.rss.exceptions.StopStepException
import com.rss.model.RssUpdate
import com.rss.utils.DislogLogger
import org.springframework.batch.item.ItemWriter

class RssSubscriptionWriter(private val messagingClient: MessagingClient) : ItemWriter<List<RssUpdate?>> {
    @Throws(Exception::class)
    override fun write(list: List<List<RssUpdate?>>) {
        if (list.isNotEmpty()) {
            for (updateList in list) {
                for (update in updateList) {
                    //logger.info("Publishing update for ${update!!.displayName} + ${update.url}")
                    messagingClient.sendRssUpdate(update!!)
                }
            }
        }
        //throw StopStepException("Test")
        //print(list)
    }
}